src/basics.md

```diff
@@ -362,6 +363,52 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-verify-extract-email]: #ex-verify-extract-email
+<a name="ex-verify-extract-email"></a>
+## Verify and extract login from an email address
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Validates that an email address is formatted correctly, and extracts everything
+before the @ symbol.
+
+```rust
+#[macro_use]
+extern crate lazy_static;
+extern crate regex;
+
+use regex::Regex;
+use std::borrow::Cow;
+
+// Validates that something has an @ symbol, followed by something that
+// looks like a domain. Returns what came before the @ symbol, if everything
+// looks valid. Prevents regex recompilation using lazy_static!
+fn extract_login(input: &str) -> Option<Cow<str>> {
+    lazy_static! {
+        static ref RE: Regex = {
+            let mut pattern = r"^([^@\s]+)@".to_owned();
+            pattern += r"([[:word:]]+\.)*[[:word:]]+$";
```

Comment:
no need to construct the regex in such way please use multiline regex (whitespace insensitive) definition syntax directly within `Regex::new()` like in 
https://brson.github.io/rust-cookbook/basics.html#run-an-external-command-and-process-stdout

any (super brief) comments would go into the regex body comments (starting with #)